### DISCUSSION TITLE
Is there any way to change the resolution on android device ?

### DISCUSSION THREAD

**Discussion by gaikoz11** [ppy/osu#20698] [Ideas]
I think if there a way it can help the Performance for android device. The default is 4ms to 20ms .. its good if there are option for resolution from 720 to 2k resolution in android version the resolution setting have a huge impact on Performance it may help also in some other platform . And if there a video backed settings in the near future . ( im just asking guys )

**Comment by peppy** [comment]
There is no current way to do this.

**Reply by heladeradragon** [reply]
You can using ADB, but its system-wide. You just have to connect it to your PC using ADB and then run `adb shell wm size [height]x[width]` (make sure USB debugging is enabled in the developer settings).
To reset it, just run `adb shell wm size reset`. However, keep in mind that this will apply the resolution change to the entire device, not just osu!

**Reply by JeromTWinL** [reply]
> You can using ADB, but its system-wide. You just have to connect it to your PC using ADB and then run `adb shell wm size [height]x[width]` (make sure USB debugging is enabled in the developer settings). To reset it, just run `adb shell wm size reset`. However, keep in mind that this will apply the resolution change to the entire device, not just osu!

Doing this will change resolution of android, but you can't use touch display with incompatible touch resolution. And causes system problems.

**Reply by peppy** [reply]
There's a [WIP](https://github.com/smoogipoo/osu-framework/tree/render-scale) implementing render scale, but it need a bit more work. We'll likely eventually see this added as a game setting.

**Comment by cheeezie** [comment]
Most games instead of changing resolution, they decrease the render resolution (example, lower texture quality and other graphical magic). It could be possible as peppy has mentioned it as a WIP. But it still has to maintain consistency with other elements like skins and UI scaling.
Even after such implementation, the display panel may still display content at its native resolution while the game's resolution is toned down.
